yarn-project/foundation/src/fields/grumpkin_scalar.ts

```diff
@@ -0,0 +1,137 @@
+import { toBigIntBE, toBufferBE, toHex } from '../bigint-buffer/index.js';
+import { randomBytes } from '../crypto/index.js';
+import { BufferReader } from '../serialize/buffer_reader.js';
+import { Fr } from './fields.js';
+
+/**
+ * Represents a field element in a prime finite field with modulus defined by the constant MODULUS.
+ * @remarks Called GrumpkinScalar because it is used to represent elements in Grumpkin's scalar field as defined in
+ *          the Aztec Yellow Paper.
+ */
+export class GrumpkinScalar {
+  static MODULUS = 0x30644e72e131a029b85045b68181585d97816a916871ca8d3c208c16d87cfd47n;
```

Comment:
Isn't this the element we already had defined as `Fq` in `yarn-project/foundation/src/fields/fields.ts`?

Reply:
Keeping it separate of `Fq` since Kev will do the fields refactor as discussed on slack.